### ISSUE TITLE
Promoting xournalpp

### ISSUE THREAD

**Issue by matzebond** [xournalpp/xournalpp#2045]
Hey all,

first let me express my gratitude towards all the contributers of this wonderful software.
xournalpp is one application that I was looking for for quite a while.
I envied my friends who could use OneNote or some iOS Software to take notes in class and while I was looking for a similar application on Linux I only found keyboard based note taking applications.
Obviously my search was not thorough enough but I still think great software as xournalpp should be found with ease.

Which brings me to me "feature request":
I want to generate some publicity for xournalpp.
I think people on Linux should not have to search to hard for a pen based note-taking app.

Some ideas to to this:
- A website, a github readme page might not generate enough visibility
- While it is in Arch Wiki [List of Applications](https://wiki.archlinux.org/index.php/List_of_applications#Graphical_21) a separate article in the Arch Wiki would be nice
- A wikipedia entry and an entry in [Comparison_of_note-taking_software](https://en.wikipedia.org/wiki/Comparison_of_note-taking_software)
- Include it as default software in some distros (e.g. Ubuntu, Linux Mint) like LibreOffice is
- Collaborate with developers of different FOSS note-taking software an give cross references
- Maybe a clear description of what xournalpp is and what it is not
- A higher ranking in https://alternativeto.net/software/microsoft-onenote/ although I don't want to register an account...

Please let me what you think could improve the visibility of xournalpp.

Cheers,
Matthias

**Comment by rolandlo** [comment]
Hi Matthias

All your ideas sound pretty good to me. Let me add some more ideas:

- Make tutorial videos (on youtube) and videos comparing xournalpp to other note taking software
- Comment about the existence and usefulness of xournalpp in the comment section of existing (youtube) videos that compare different note taking software / OneNote alternatives  
- Complete the [user manual](https://github.com/xournalpp/xournalpp/wiki/User-Manual) and add more details
- Suggest xournalpp as a topic to bloggers and vloggers (I actually learned about xournalpp from [this article ](https://www.omgubuntu.co.uk/2019/06/xournalpp-handwritten-notes-app-linux) on omgubuntu)

Best wishes, 
Roland

**Comment by Technius** [comment]
In my opinion, the top priority is getting an official website up (e.g. `xournalpp.github.io`) and then filling out the user guide so we can address discoverability.

**Comment by pktiuk** [comment]
Yes, visibility is a problem, I was looking for an app to easily make notes on my PDF-s since several months and I couldn't find anything else than Foxit Reader (which is not very good on linux) and I have found it two days ago accidentally.
I was looking for it mainly by typing: `making notes on PDF` etc. and by checking alternatives to onenote on [alternativeto.net](alternativeto.net) and I haven't found it there.

Moreover I couldn't find any comparison of xournal and xournal++ (It would be nice to have it described in README, which would be more readable with table of contents), currently I just assume xournal++ is alive fork of fead xournal.
Moreover I have to mention, that in terms of availability on linux there is no xournal++ in debian repository, but there is old xournal. Many people on Debians and Ubuntu wij just choose old xournal, because they can instal it much easier. On top of that .deb on release site doesn't work on Ubuntu 20.04, so I had to download flatpak package.

> In my opinion, the top priority is getting an official website up (e.g. xournalpp.github.io) and then filling out the user guide so we can address discoverability.

I think for now simple [docsify](https://docsify.js.org/#/) site would be enough for basic instructions and informations.

**Comment by Technius** [comment]
> Moreover I couldn't find any comparison of xournal and xournal++ (It would be nice to have it described in README, which would be more readable with table of contents), currently I just assume xournal++ is alive fork of fead xournal.

This is a good point--we'll have to keep this in mind when we make the website.

> Moreover I have to mention, that in terms of availability on linux there is no xournal++ in debian repository, but there is old xournal. Many people on Debians and Ubuntu wij just choose old xournal, because they can instal it much easier. On top of that .deb on release site doesn't work on Ubuntu 20.04, so I had to download flatpak package.

There's a Debian maintainer in the community working on this, but I don't know what the current status is.

> > In my opinion, the top priority is getting an official website up (e.g. xournalpp.github.io) and then filling out the user guide so we can address discoverability.
>
> I think for now simple [docsify](https://docsify.js.org/#/) site would be enough for basic instructions and informations.

I'm personally of the opinion that simple websites, like documentation sites, should only use pure HTML/CSS. We've been considering setting up a very simple website based on https://www.mkdocs.org/ or https://readthedocs.org/. Should be trivial with GitHub pages and GitHub actions.

**Comment by matzebond** [comment]
I created a [draft Wikipedia page](https://en.wikipedia.org/wiki/Draft:Xournalpp) and added Xournalpp to [Comparison of note-taking software](https://en.wikipedia.org/wiki/Comparison_of_note-taking_software).
I would also like to help with a website. mkdocs or readthedocs sound good to me but GitHub actions give me headaches so maybe some one else should do setup :-|

> There's a Debian maintainer in the community working on this, but I don't know what the current status is.

From what I have read it's a rather long process to get into the Debian repo.

**Comment by rolandlo** [comment]
Good job with the wiki page @matzebond! One minor thing: infinity pages are not yet implemented. So you may want to delete the phrase "and infinity mode" from the wiki page.

**Comment by pktiuk** [comment]
@matzebond 
I think it would be good to add an icon of app and example screen from app window to this site on Wikipedia.
It would be more attractive for people who are visualizers like me :smile: 

You could get inspiration from [LibreOffice Wikipedia page](https://en.wikipedia.org/wiki/LibreOffice)

**Comment by matzebond** [comment]
I removed "and infinity mode" but hopefully I can add it again soon :-)
I also added the icon and screenshot from the readme. Looks much better now :+1:

**Comment by pktiuk** [comment]
> From what I have read it's a rather long process to get into the Debian repo.

It takes longer than I expected. :hourglass_flowing_sand: 
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=927076
But on the other side xournalpp still doesn't have working .deb package.
When we will have, then we could inform about it there.

**Comment by daholzfeind** [comment]
> I'm personally of the opinion that simple websites, like documentation sites, should only use pure HTML/CSS. We've been considering setting up a very simple website based on https://www.mkdocs.org/ or https://readthedocs.org/. Should be trivial with GitHub pages and GitHub actions.

On the one hand maybe Jekyll (https://jekyllrb.com/) is a good solution for this purposes. 

On the other hand, if you only want a website with just docs, Sphinx (https://www.sphinx-doc.org/en/master/) would be a good choice.

**Comment by rolandlo** [comment]
Thanks for your suggestions @daholzfeind.
The Jekyll website showcases some pretty nice examples ob websites created with Jekyll. A website of that kind would surely attract more people using xournalpp. The creation of websites with Jekyll doesn't look too complicated, so all in all it would be my favorite choice from all suggestions so far.

**Comment by daholzfeind** [comment]
@rolandlo Maybe I can help, creating the website. Should I open a Follow Up Issue/PR?

**Comment by rolandlo** [comment]
@daholzfeind 
You are very welcome to help with the website. Just go ahead creating a PR, so we can see how a xournalpp website with Jerkyll might look like and discuss (in a follow up issue) pros and cons of your solution with regards to potential solutions using other website creaters. 
I must add that I can only express my own opinion and am not aware of considerations made by @Technius and other core devs.

**Comment by Technius** [comment]
I have set up the website repository on https://github.com/xournalpp/xournalpp.github.io. It uses MkDocs to reduce the barrier to entry and [it autodeploys](https://xournalpp.github.io/) whenever the default branch (`site`) is updated. Feel free to submit PRs to improve the website.

**Comment by matzebond** [comment]
My Wikipedia article was rejected for now. I think they want more sources. I will add the article rolandlo posted

**Comment by rolandlo** [comment]
@matzebond The OMG article and [a review in the Linux User magazine](https://www.linux-community.de/ausgaben/linuxuser/2019/05/schnell-notiert/) (in German) are already included on the wikipedia draft page. Two more independent references would be: [this Ubuntuhandbook post](http://ubuntuhandbook.org/index.php/2019/10/xournal-1-0-5-released-install-ubuntu/) (in English) and [this Linux Bibel review](https://www.linux-bibel-oesterreich.at/2020/04/24/xournal-xournalapp-notizen-am-touchscreen-unter-linux/) (in German). 
You better check carefully, if the references qualify as "independent reliable sources" (with a reputation for fact-checking and oversight), before submitting the draft again. You may want to ask for live help from experienced editors to determine that. Also note that website citations should have a date when the website was accessed, and a translation of the title should be included (as described on [this site](https://en.wikipedia.org/wiki/Template:Cite_web#Foreign_language_and_translated_title)).

**Comment by pktiuk** [comment]
It would be good to add Xournal++ to [sourceforge.net](sourceforge.net) and edit site of original Xournal, which is still [quite popular](https://sourceforge.net/projects/xournal/files/stats/timeline) there (about 11.000 downloads in April).

**Comment by matzebond** [comment]
With some help of the community the Wikipedia article was improved and submitted again. I think we still might not have enough "secondary" sources but lets see.
@rolandlo I thinks those blog posts are not suitable for Wikipedia because the are "self-published sources"

**Comment by rolandlo** [comment]
Ok, let's see if the improved article is accepted. It definitively looks quite a bit more professional now.
A side note: The French Ubuntu documentation has recently updated its entry on Xournal by replacing Xournal with Xournal++, see [this link](http://doc.ubuntu-fr.org/xournal).

**Comment by rolandlo** [comment]
@matzebond 
What about this [softpedia review](https://www.softpedia.com/get/Multimedia/Graphic/Graphic-Editors/Xournal.shtml)? Looks like a reliable source, worth of inclusion in the wikipedia article, doesn't it?

**Comment by rolandlo** [comment]
@matzebond 
Another reliable source (although in German) might be the [Chip.de article](https://praxistipps.chip.de/home-office-diese-software-brauchen-sie_118300) on "software needed in home office", where xournal/xournalpp took second place right after LibreOffice.

**Comment by rolandlo** [comment]
@matzebond 
Another source which might be considered as "reliable", is the [Linux Uprising article](https://www.linuxuprising.com/2019/10/handwritten-notes-and-pdf-annotation.html)

**Comment by pktiuk** [comment]
@matzebond 
I think you should edit your first message to make it more informative for new people in this thread
You should split these points with ideas to three categories:

what you can do now (even if you are newcomer)
- like xournal++ on alternativeto (with link to site) 
- help with writing guides on website (and put here link to site and it's repo)
-...

Checklist with things to do (with checkboxes- things which can be easily marked as "done" ):

- [ ]  A wikipedia entry and an entry in Comparison_of_note-taking_software
- [ ] add xournal++ to official ( [debian](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=927076),arch,fedora...) repositories (I think this point should get its own issue and link to it should be put here) - this step would make xournalpp more accessible 
- [ ] Complete the user manual
- [ ] ...

Other ideas:
- Suggest xournalpp as a topic to bloggers and vloggers
- everything not in previous categories...

It would be nice to add link (where such al link exists) to every of these points (it would make it easier to navigate and check progress)

**Comment by matzebond** [comment]
@pktiuk the first post should be more overviewy now :)

@rolandlo thanks for all the links. I would like to wait for the decision before working further on the article. I for my part have seen  worse software articles on Wikipedia.

**Comment by pktiuk** [comment]
I think, that adding xournalpp to Debian repositories should have it's own issue, so I have created it. https://github.com/xournalpp/xournalpp/issues/2148

**Comment by rolandlo** [comment]
@matzebond You have probably noticed that the wikipedia article has been rejected again (3 months ago). The reviewer comments: 

> articles about how to use the app do not establish notability. Source must be independent, secondary sources that give significant attention to the subject.

I don't quite get what they expect. Let me mention nevertheless, that the well known `Chip.de` regards Xournal++ highly, see [this article](https://www.chip.de/news/Windows-10-PDF-Notizen-Kostenlose-Top-Tools_182745544.html) and [that one](https://www.chip.de/downloads/Xournal_183156091.html) and the one I mentioned in a previous post.

**Comment by matzebond** [comment]
@rolandlo 
Yes I noticed. The rejection was harsh and it feels like xournalpp would only become an "established" software project in the eyes of Wikipedia, if the NewYork Times writes a 5-page article about it...

**Comment by rolandlo** [comment]
What about trying the German version then? Seems like they have a bit lower requirements. There is already an article about Xournal (without any references except the user manual and the website of Xournal++)

**Comment by matzebond** [comment]
Sound like a good idea. I will look into creating a draft for the German
Wikipedia and hopefully this will also make acceptance of the English
version more likely.

**Comment by rolandlo** [comment]
By the way I think we should call our application `Xournal++` on Wikipedia (not `Xournalpp`) and refer to `xournalpp` as the package name.

**Comment by barak** [comment]
I've whipped the Debian packaging into shape, see https://bugs.debian.org/956272 and branch `debian` in https://github.com/barak/xournalpp, _will upload into Debian shortly_. The last sticking point was getting `debian/copyright` correct for all the misc bits and bobs. I hope I have it correct enough for the Debian ftpmasters to accept the package. But, I would appreciate it if someone more familiar with things could give it a once-over and correct any errors. The providence of the various SVG files was particularly difficult to trace. And that one hashing include file from boost: who actually wrote that? And when?

Pull requests against `debian/copyright` on my branch are particularly welcome!

**Comment by rolandlo** [comment]
@barak Thank you for your efforts. 
About the copyright: In #2148 I have tried to collect all information on the files that had a "God knows" mark previously. Only for two icons (`snapping.svg` and `grid_snapping.svg` added by @morrolinux) I didn't get an answer. 
The hashing include file from boost was added by @Febbe in https://github.com/xournalpp/xournalpp/commit/f4e4fcff86611e76a391f41900199a7bd58a14e5. It's a shortened version of the original file, it seems (compare https://www.geometrictools.com/GTE/Mathematics/HashCombine.h).

I think the list of Xournal++ team members should include my name (and Email address roland_loetscher@hotmail.com), since it is included in the About dialog for version 1.0.20 (to which I have contributed).

**Comment by barak** [comment]
@rolandlo Thanks, that's fantastic. I was unaware of #2148, would have commented there if I'd known. Will shift this discussion over.

I've already uploaded to Debian (freeze is already underway, so in a bit of a rush to slip it in) so these debian/copyright fixes will go into the next upload.

As a general comment, because the file isn't functional, transcription errors etc can easily slip in. Especially if it's me doing the editing based on my rushed interpretation of comments by other people. Pull requests or patches or whatever avoid this...

**Comment by matzebond** [comment]
I have translated the Wikipedia article for the [german Wiki](https://de.wikipedia.org/wiki/Xournal%2B%2B). There is no draft process but they still want some changes ("Wikifizieren") otherwise the article might get deleted. There is now also an entry in [Wikidata](https://www.wikidata.org/wiki/Q62704288) so at some point it should be enough evidence for accepting into the english Wiki.

**Comment by sven-frotscher** [comment]
As for sources meeting Wikipedia's notability criteria, I learned about this project from a [German Linux magazine](https://www.linux-user.de/Downloads/LUCE/2019/lu-ce_2019-05.pdf). When I find the real thing and not a stripped down online version (which sadly does not contain the article of interest), I can provide more info, perhaps even a scan of the article.

**Comment by rolandlo** [comment]
@sven-frotscher Isn't this the same as [this article](https://www.linux-community.de/ausgaben/linuxuser/2019/05/schnell-notiert/), which is already referenced on the Wikipedia page?

**Comment by sven-frotscher** [comment]
@rolandlo You're right, my bad.
I cleaned up the citation on the Wiki page a bit (because the article has a German title).

**Comment by rolandlo** [comment]
The German [wiki page](https://de.wikipedia.org/wiki/Xournal%2B%2B) on Xournal++ is a bit outdated, since it claims that 1.0.18 is the latest version. Would someone be willing to update it to the newly released version 1.1.0 / 1.1.0-hotfix-1?

**Comment by matzebond** [comment]
Done :)

**Comment by rolandlo** [comment]
Thanks @matzebond 
It would be nice, if the icon and screenshot were updated as well. Compare [flathub](https://flathub.org/apps/details/com.github.xournalpp.xournalpp). The [icon](https://github.com/xournalpp/xournalpp/tree/master/ui/pixmaps) and [screenshots](https://github.com/xournalpp/xournalpp-examples/tree/main/appdata/screenshots) are available in the repo.

**Comment by matzebond** [comment]
@rolandlo could you please specify the copyright (ideally CC BY-SA 4.0) of https://github.com/xournalpp/xournalpp-examples then I can import the images to wikimedia.

**Comment by rolandlo** [comment]
@matzebond I have added the CC-BY-SA-4.0 license to the `xournalpp-examples` repo, see https://github.com/xournalpp/xournalpp-examples

**Comment by matzebond** [comment]
Thanks. The german wikipedia page now shows the new logo and picture. Btw the english wiki draft page got deleted because it was stale for 6 months. We could request an [undeletion](https://en.wikipedia.org/wiki/Wikipedia:REFUND/G13) but I am currently not interested in improving and re-resubmitting it.

**Comment by Fatih20** [comment]
- [x]  Improved the homepage to entice new visitors to use it.

**Comment by lrusso96** [comment]
@matzebond there is also [an article on it.wikipedia](https://it.wikipedia.org/wiki/Xournal%2B%2B).
